### PR TITLE
Offhours holidays

### PR DESCRIPTION
**Pull request by hav3wb**

### CODE REVIEW

**Comment by hav3wb** [comment]
Allows for submitting a list of holidays to exclude from either offhours or onhours. Can also specify a url containing the list using the same ValueFrom syntax. I also made sure to include additional tests, a validation check to prevent users from supplying two lists and an updated description provided by the `custodian schema ec2.filters.offhour` command.

**Comment by hav3wb** [comment]
For the sake of consistency, I used dates of the format YYYY-MM-DD to check against as that was the default provided by python datetime. I opted to omit an extra variable to set the syntax of the list as I thought it was unnecessary, but my code could be reformatted to require a variable dictating the syntax of the list (i.e. syntax: "%Y-%m-%d"). Finally, I included year in all of the holidays (must use 2017-12-25 instead of 12-25) because several of the US holidays are celebrated on different days each year. This could cause issues if the list provided is not updated yearly by shutting down instances on the previous year's holiday dates. This way, if it is not updated, then nothing gets accidentally skipped.

File: c7n/filters/offhours.py [hunk 1]
```diff
@@ -376,6 +382,14 @@ def process_resource_schedule(self, i, value, time_type):
             return False
         now = datetime.datetime.now(tz).replace(
             minute=0, second=0, microsecond=0)
+        now_str = now.date().strftime("%Y-%m-%d")
```
Comment (kapilt) [inline c7n/filters/offhours.py]: the date() here is redundant, can just strftime on the datetime

File: c7n/filters/offhours.py [hunk 2]
```diff
@@ -235,6 +236,8 @@ class Time(Filter):
             'weekends': {'type': 'boolean'},
             'weekends-only': {'type': 'boolean'},
             'opt-out': {'type': 'boolean'},
+            'skip-days': {'type': 'array', 'items': {'type': 'string'}},
```
Comment (kapilt) [inline c7n/filters/offhours.py]: can you do a regex constraint on the string value for skip-days to ensure the format your expecting.

File: c7n/filters/offhours.py (hunk 1 above)
Comment (hav3wb) [reply c7n/filters/offhours.py]: Yep, I just used date to make it clear I was only getting date, but I will switch in my next commit.

File: c7n/filters/offhours.py (hunk 2 above)
Comment (kapilt) [reply c7n/filters/offhours.py]: examples https://spacetelescope.github.io/understanding-json-schema/reference/regular_expressions.html

Comment (hav3wb) [reply c7n/filters/offhours.py]: @kapilt Added!

**Comment by jantman** [comment]
Looks great to me, and is also very timely as we'd been discussing this at my employer lately (as well as how to skip termination on holidays, but that's a much larger scope....).

One request though; do you think you could add some documentation about this to the docstring at the top of [c7n/filters/offhours.py](https://github.com/capitalone/cloud-custodian/blob/master/c7n/filters/offhours.py#L14)? I see that you added some docs to the EC2-specific code, but the large docstring at the top of [offhours.py](https://github.com/capitalone/cloud-custodian/blob/master/c7n/filters/offhours.py#L14) is used (via the sphinx [automodule directive](https://github.com/capitalone/cloud-custodian/blob/master/docs/source/quickstart/offhours.rst)) to generate the [Example offhours policy](http://www.capitalone.io/cloud-custodian/docs/quickstart/offhours.html) page, which is the most complete documentation on the use of on/off hours.

**Comment by hav3wb** [comment]
@jantman I updated the offhours documentation and added a separate section for dealing with holidays.

**Comment by jantman** [comment]
Thanks so much for adding the docs!

**Pull request merged**